sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -410,6 +410,32 @@ class DateTimeUtilsSuite extends SparkFunSuite {
     assert(getDayInYear(getInUTCDays(c.getTimeInMillis)) === 78)
   }
 
+  test("SPARK-26002: correct day of year calculations for Julian calendar years") {
```

Comment:
can we have an end-to-end test?

Reply:
@cloud-fan 
Do you mean adding a new test to [datetime.sql](https://github.com/apache/spark/blob/421673a4f6ae8820b57bb1fb92d9fe03dc646cde/sql/core/src/test/resources/sql-tests/inputs/datetime.sql)? 

What about adding the ones I used as a manual testing (from the PR description)?

Reply:
SGTM